src/gadgets.rs

```diff
@@ -0,0 +1,102 @@
+//! MPC circuits implemented directly in the library as gadgets for more complex MPC operations
+
+use std::iter;
+
+use ark_ec::CurveGroup;
+use itertools::Itertools;
+
+use crate::{algebra::AuthenticatedScalarResult, MpcFabric};
+
+/// A prefix product gadget, computes the prefix products of a vector of values, where
+/// for `n` values, the `i`th prefix product is defined as:
+///     x0 * x1 * ... * xi
+///
+/// The method used here is that described in:
+///     https://dl.acm.org/doi/pdf/10.1145/72981.72995
```

Comment:
Just for clarity, can we also mention that this specifically is an instance of the secret matrix multiplication protocol in section 4.2 of the paper?